.circleci/config.yml

```diff
@@ -16,26 +16,61 @@ jobs:
       - checkout
 
       - restore_cache:
-          key: oppia-tools-cache
+          keys:
+            - v1.0-oppia-tools-cache-{{ checksum "oppia-tools-dependency-list.txt" }}
 
       - run:
-          name: install dependencies
-          command: |
-            bash scripts/setup.sh
-            bash scripts/setup_gae.sh
-
-      - run:
-          name: run tests
+          name: Run tests
           command: |
             bash scripts/run_backend_tests.sh --generate_coverage_report
 
+      - save_cache:
+          key: v1.0-oppia-tools-cache-{{ checksum "oppia-tools-dependency-list.txt" }}
+          paths:
+            - ./oppia_tools/
+  linter:
+    docker:
+      # specify the version you desire here
+      # use `-browsers` prefix for selenium tests, e.g. `3.6.1-browsers`
+      - image: circleci/python:2.7.15-browsers
+
+    working_directory: ~/foundation-website
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1.0-oppia-node-modules-cache-{{ checksum "package-lock.json"}}
+      - restore_cache:
+          keys:
+            - v1.0-oppia-tools-cache-{{ checksum "oppia-tools-dependency-list.txt" }}
+      - restore_cache:
+          keys:
+            - v1.0-oppia-pip-cache-{{ checksum "ci-linter-requirements.txt" }}
+
       - run:
-          name: run linter
+          name: Run linter
           command: |
             bash scripts/run_linter.sh
 
       - save_cache:
-          key: oppia-tools-cache
+          key: v1.0-oppia-tools-cache-{{ checksum "oppia-tools-dependency-list.txt" }}
```

Comment:
For ease of readability consider putting these in the same order as restore_cache.

That said -- why three stanzas instead of one? I know that each can be updated independently but I wonder if it's worth the effort and corresponding complexity in the config file (it doesn't take that long to regenerate, right?)

Reply:
1) I have put the keys in `restore_cache` and `save_cache` in the same order.

2) If I have a different key for a different filepath, I would need a separate stanza, thus the need for multiple stanzas.